fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -1066,11 +1072,16 @@ private FHIRPersistenceNotSupportedException buildNotSupportedException(String m
                     .text(string("Invalid page size: " + pageSize))
                     .build())
                 .build());
-            pagingContext.setPageSize(10);
+            // Pick a valid default if lenient
+            if (pagingContext.isLenient()) {
+                pagingContext.setPageSize(10);
+            }
         }
 
-        int lastPageNumber = Math.max(((pagingContext.getTotalCount() + pageSize - 1) / pageSize), 1);
-        pagingContext.setLastPageNumber(lastPageNumber);
+        if (pagingContext.getTotalCount() != null) {
+            pagingContext.setLastPageNumber(Math.max(((pagingContext.getTotalCount() + pageSize - 1) / pageSize), 1));
```

Comment:
Separate function so we can have a unit-test for the math?

Reply:
Will do.

Reply:
Updated tests in AbstractPagingTest to check the math.